src/luaconf.h

```diff
@@ -123,6 +123,16 @@
 #define LUA_INTFRMLEN		"l"
 #define LUA_INTFRM_T		long
 
+/* Allows for aligning the userdata struct. Valid values are 8 or 16. Default is 8.
+*/
+#if !defined(LUA_USERDATA_ALIGNMENT)
+  #define LUA_USERDATA_ALIGNMENT 8
+#endif
```

Comment:
Might be overkill to have this as optional.
One option could be to simply change it to 16 bytes.
However, I'm unclear if that would break the ABI for existing byte code? 🤔 

That's why I opted for this setting, so that the developer can choose.